src/server/checker/CheckerService.ts

```diff
@@ -164,6 +181,97 @@ export class CheckerService {
     return checker
   }
 
+  private findAndCheckAuthOwner: (
+    id: string,
+    user: User | null
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ) => Promise<(Model<Checker, any> & Checker) | null> = async (id, user) => {
+    const checker = await this.CheckerModel.findByPk(id, {
+      include: [this.UserModel],
+    })
+    if (checker) {
+      const isAuthorized = checker.users?.some(
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        (u) => u.UserToChecker.isOwner && u.id === user?.id
+      )
+      if (!isAuthorized) {
+        throw new Error('Unauthorized')
+      }
+    }
+    return checker
+  }
+
+  listCollaborators: (id: string, user: User) => Promise<CollaboratorUser[]> =
+    async (id, user) => {
+      const checker = await this.findAndCheckAuth(id, user)
+      if (checker) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return checker.users
+      } else return []
+    }
+
+  addCollaborator: (
+    id: string,
+    user: User,
+    collaboratorEmail: string
+  ) => Promise<void> = async (id, user, collaboratorEmail) => {
+    const transaction = await this.sequelize.transaction()
+    const options = this.isSqliteFile ? {} : { transaction }
+
+    try {
+      const checker = await this.findAndCheckAuth(id, user)
+      collaboratorEmail = collaboratorEmail.toLowerCase()
+
+      if (checker) {
+        const collaboratorUser = await this.UserModel.findOne({
+          where: { email: collaboratorEmail },
+        })
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        if (checker.users?.some((u) => u.email === collaboratorEmail))
+          throw new Error('User is already a collaborator')
+        if (collaboratorUser) {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          await collaboratorUser.addChecker(
+            checker,
+            { through: { isOwner: false } },
+            options
+          )
+          await transaction.commit()
+        } else {
+          throw new Error('No such user')
+        }
+      }
+    } catch (error) {
+      this.logger?.error(error)
+      await transaction.rollback()
+      throw error
+    }
+  }
+
+  deleteCollaborator: (
+    id: string,
+    user: User,
+    collaboratorEmail: string
+  ) => Promise<void> = async (id, user, collaboratorEmail) => {
+    const checker = await this.findAndCheckAuth(id, user)
+    if (checker) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const collaborator = checker.users?.find(
```

Comment:
If `removeUser` returns a value, we can potentially use that to determine if the collaborator exists in the first place. Otherwise, similarly, we should wrap this up in a transaction.